src/gui/menu_list.cpp

```diff
@@ -0,0 +1,24 @@
+#include "gui/menu_item.hpp"
+#include "gui/menu_list.hpp"
+#include "gui/menu_manager.hpp"
+
+ListMenu::ListMenu(std::vector<std::string>* items_, int* selected_) : 
+  items(items_), 
+  selected(selected_)
+{
+  int i = 0;
+  for(auto& item : *items) {
+    add_entry(i, item);
+    i++;
+  }
```

Comment:
Less important, but might be good to consider: what are the tradeoffs of this vs. a regular `for (int i = 0; i < items->size(); i++)` loop?

Reply:
Not much. I'll replace this too.